### PR TITLE
dzhyv/Return to home page test

### PR DESCRIPTION
**Pull request by dzhyv**

### CODE REVIEW

File: src/test/java/com/ss/ita/rozetka/rozetka/clickRozetkaLogoTest.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+import com.ss.ita.rozetka.ui.pages.HomePage;
+import com.ss.ita.rozetka.ui.ProductsEnum.GeneralProductCategory;
+import org.testng.Assert;
+import  org.testng.annotations.Test;
+
+import static com.ss.ita.rozetka.ui.ProductsEnum.GeneralProductCategory.PRODUCTS_FOR_HOUSE;
+import static com.ss.ita.rozetka.ui.ProductsEnum.ProductCategoryAndSubCategory.GARDEN_TECH_CATEGORY;
+
+public class clickRozetkaLogoTest {
```
Comment (toleksyn) [inline src/test/java/com/ss/ita/rozetka/rozetka/clickRozetkaLogoTest.java:11]: java code convention

File: src/test/java/com/ss/ita/rozetka/rozetka/clickRozetkaLogoTest.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+    @Test
+    public void verifyClickRozetkaLogo(){
+        String actualUrl = new HomePage().open()
+                .openProductCategoryPage(PRODUCTS_FOR_HOUSE.getName())
```
Comment (toleksyn) [inline src/test/java/com/ss/ita/rozetka/rozetka/clickRozetkaLogoTest.java:16]: re-do openProductCategoryPage() to work with enum directly

File: src/test/java/com/ss/ita/rozetka/rozetka/clickRozetkaLogoTest.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import org.testng.Assert;
+import  org.testng.annotations.Test;
+
+import static com.ss.ita.rozetka.ui.ProductsEnum.GeneralProductCategory.PRODUCTS_FOR_HOUSE;
+import static com.ss.ita.rozetka.ui.ProductsEnum.ProductCategoryAndSubCategory.GARDEN_TECH_CATEGORY;
+
+public class clickRozetkaLogoTest {
+
+    @Test
+    public void verifyClickRozetkaLogo(){
```
Comment (toleksyn) [inline src/test/java/com/ss/ita/rozetka/rozetka/clickRozetkaLogoTest.java:14]: where does it click on logo in test?

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/ss/ita/rozetka/ui/pages/HeaderPage.java [hunk 4]
```diff
@@ -3,5 +3,5 @@
 import com.ss.ita.rozetka.ui.elements.Header;
 
 public abstract class HeaderPage {
-    protected Header header = new Header();
+    public Header header = new Header();
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/ui/pages/HeaderPage.java:6]: no public fields in PO

File: src/test/java/com/ss/ita/rozetka/ui/NavigatingToHomePageTest.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import static com.ss.ita.rozetka.ui.ProductsEnum.GeneralProductCategory.PRODUCTS_FOR_HOUSE;
+import static com.ss.ita.rozetka.ui.ProductsEnum.ProductCategoryAndSubCategory.KITCHEN_APPLIANCES_CATEGORY;
+
+public class NavigatingToHomePageTest extends TestRunner {
+
+    @Test
+    public void verifyReturnToHomePage() {
+        String actualUrl = new HomePage().open()
+                .openProductCategoryPage(PRODUCTS_FOR_HOUSE)
+                .openProductTypePage(KITCHEN_APPLIANCES_CATEGORY)
```
Comment (toleksyn) [inline src/test/java/com/ss/ita/rozetka/ui/NavigatingToHomePageTest.java:18]: how do we know this page opened at all?

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/test/java/com/ss/ita/rozetka/ui/NavigatingToHomePageTest.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+                .getHeader()
+                .getCurrentUrl();
+        Assert.assertEquals(productSubcategoryUrl, "https://bt.rozetka.com.ua/tehnika-dlya-kuhni/c435974/");
+        String homePageUrl = page
+                .getHeader()
+                .openHomePage()
+                .getHeader()
+                .openHomePage()
+                .getHeader()
```
Comment (toleksyn) [inline src/test/java/com/ss/ita/rozetka/ui/NavigatingToHomePageTest.java:33]: three times getHeader()

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/test/java/com/ss/ita/rozetka/ui/NavigatingToHomePageTest.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+import com.ss.ita.rozetka.ui.TestUtils.TestRunner;
+
+import org.testng.annotations.Test;
+
+import static com.ss.ita.rozetka.ui.ProductsEnum.GeneralProductCategory.PRODUCTS_FOR_HOUSE;
+import static com.ss.ita.rozetka.ui.ProductsEnum.ProductCategoryAndSubCategory.KITCHEN_APPLIANCES_CATEGORY;
+import static com.ss.ita.rozetka.ui.util.PageUtil.getCurrentUrl;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class NavigatingToHomePageTest extends TestRunner {
```
Comment (toleksyn) [inline src/test/java/com/ss/ita/rozetka/ui/NavigatingToHomePageTest.java:14]: NavigationTest

File: src/test/java/com/ss/ita/rozetka/ui/NavigatingToHomePageTest.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+        ProductCategoryPage page = new HomePage()
+                .open()
+                .openProductCategoryPage(PRODUCTS_FOR_HOUSE);
+        assertThat(getCurrentUrl())
+                .as("Products for house category page should be opened")
+                .isEqualTo("https://bt.rozetka.com.ua/");
+        page.openProductTypePage(KITCHEN_APPLIANCES_CATEGORY);
+        assertThat(getCurrentUrl())
+                .as("Kitchen appliances category page should be opened")
+                .isEqualTo("https://bt.rozetka.com.ua/tehnika-dlya-kuhni/c435974/");
```
Comment (toleksyn) [inline src/test/java/com/ss/ita/rozetka/ui/NavigatingToHomePageTest.java:27]: checking that page opened only by URL is not fool proof and never enough

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/ss/ita/rozetka/ui/pages/HomePage.java [hunk 9]
```diff
@@ -20,4 +20,9 @@ public ProductCategoryPage openProductCategoryPage(GeneralProductCategory catego
         $x(format("//div[contains(@class,'menu-wrapper_state_static')]/descendant::a[contains(@href,'%s')]", category.getName())).click();
         return new ProductCategoryPage();
     }
+
+    @Step("HomePage: verify that home page slider is visible")
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/ui/pages/HomePage.java:24]: this is not verification

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/ss/ita/rozetka/ui/pages/HomePage.java [hunk 10]
```diff
@@ -21,6 +22,11 @@ public ProductCategoryPage openProductCategoryPage(GeneralProductCategory catego
         return new ProductCategoryPage();
     }
 
+    @Step("HomePage: get home page slider visibility status")
+    public boolean isSliderVisible() {
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/ui/pages/HomePage.java:26]: isOpened(), then it will read as homePage.isOpened()

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/test/java/com/ss/ita/rozetka/test/NavigationTest.java [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+import static com.ss.ita.rozetka.pageobject.product.ProductCategoryAndSubCategory.KITCHEN_APPLIANCES_CATEGORY;
+import static com.ss.ita.rozetka.pageobject.utils.PageUtil.getCurrentUrl;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class NavigationTest extends TestRunner {
+
+    @Test
+    @Issue("LVTAQC630-7")
+    public void verifyReturnToHomePage() {
+        ProductCategoryPage page = new HomePage()
```
Comment (toleksyn) [inline src/test/java/com/ss/ita/rozetka/test/NavigationTest.java:20]: page -> productCategoryPage

**Review by toleksyn — CHANGES_REQUESTED** [review]

**Review by anastasia8755 — APPROVED** [review]

**Review by NazarYakh — APPROVED** [review]

**Pull request merged**